homeassistant/components/daikin/__init__.py

```diff
@@ -167,16 +185,6 @@ def _update_unique_id(entity_entry: er.RegistryEntry) -> dict[str, str] | None:
                     merge_connections=new_connections,
                 )
 
-        if device_entry.name is None:
-            _LOGGER.debug(
-                "Migrating device name to %s",
-                new_name,
-            )
-            dev_reg.async_update_device(
-                device_entry.id,
-                name=new_name,
-            )
-
```

Comment:
This was causing problems for some users as it renamed some entities.

Reply:
OK, understood 👍 

Please update the PR title + description then to explain this PR does not only removes duplicated devices but also reverts unwanted renaming of devices or move the removal of the renaming to a separate PR (this would be the preferred option).

Reply:
I'll move it to a separate PR.